src/lib/AjaxAdaptorBase.js

```diff
@@ -1,5 +1,8 @@
+import assign from 'object-assign'
+
 export default class AjaxAdaptorBase {
   serverBase() { return '' } // default will use the relative path
   pathRoot() { return '/' }
   defaultHeaders() { return {} } // any headers that should be in every ajax request
+  headers(additionalHeaders) { return assign(this.defaultHeaders(), additionalHeaders) }  // Add other headers or overwrite defaults.
```

Comment:
this is the right idea (order is importnat) -- but the assign should happen in stateham otherwise, everyone has to know this implementation detail. (since you have to create your own adaptor to use this sucker. )